replaced  the autocorrelation calculation with a proper FFT based one

wdpozzo opened the pull request.

pep8speaks commented:
Hello @wdpozzo! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`cpnest/cpnest.py`](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py):

> [Line 71:1](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L71): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 74:1](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L74): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 97:32](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L97): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 97:34](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L97): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 158:25](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L158): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 158:39](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L158): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 158:41](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L158): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 175:35](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L175): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 175:47](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L175): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 175:49](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L175): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 197:35](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L197): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 197:47](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L197): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 197:49](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L197): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 231:24](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L231): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 233:24](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L233): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 310:1](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L310): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 315:80](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L315): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (107 > 79 characters)
> [Line 315:86](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L315): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 315:99](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L315): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 315:101](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L315): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 316:77](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L316): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 316:80](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L316): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (84 > 79 characters)
> [Line 317:1](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L317): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 318:80](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L318): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 325:45](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L325): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 328:36](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L328): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 350:21](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L350): [E221](https://duckduckgo.com/?q=pep8%20E221) multiple spaces before operator
> [Line 353:80](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L353): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (127 > 79 characters)
> [Line 353:105](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L353): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 353:118](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L353): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 353:120](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L353): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 354:77](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L354): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 354:80](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L354): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (84 > 79 characters)
> [Line 355:1](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L355): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 360:45](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L360): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 363:36](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L363): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 365:1](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L365): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 371:24](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L371): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 374:26](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L374): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 376:15](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L376): [E221](https://duckduckgo.com/?q=pep8%20E221) multiple spaces before operator
> [Line 383:52](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L383): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 383:54](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L383): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 384:45](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L384): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 384:47](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L384): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 384:80](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L384): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (106 > 79 characters)
> [Line 385:44](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L385): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 385:46](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L385): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 385:80](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L385): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (103 > 79 characters)
> [Line 386:40](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L386): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 386:42](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L386): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 386:67](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L386): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 386:80](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L386): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (98 > 79 characters)
> [Line 391:80](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L391): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (92 > 79 characters)
> [Line 393:80](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L393): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (92 > 79 characters)
> [Line 394:30](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L394): [E221](https://duckduckgo.com/?q=pep8%20E221) multiple spaces before operator
> [Line 394:80](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L394): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (89 > 79 characters)
> [Line 397:30](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L397): [E221](https://duckduckgo.com/?q=pep8%20E221) multiple spaces before operator
> [Line 403:67](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L403): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 403:80](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/cpnest.py#L403): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (81 > 79 characters)

* In the file [`cpnest/nest2pos.py`](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/nest2pos.py):

> [Line 7:1](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/nest2pos.py#L7): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '
> [Line 177:9](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/nest2pos.py#L177): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 178:8](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/nest2pos.py#L178): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 179:7](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/nest2pos.py#L179): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 181:7](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/nest2pos.py#L181): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 182:7](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/nest2pos.py#L182): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 183:9](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/nest2pos.py#L183): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 186:1](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/nest2pos.py#L186): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1
> [Line 190:6](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/nest2pos.py#L190): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 191:6](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/nest2pos.py#L191): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 193:17](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/nest2pos.py#L193): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 193:33](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/nest2pos.py#L193): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 194:10](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/nest2pos.py#L194): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 195:10](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/nest2pos.py#L195): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 198:1](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/nest2pos.py#L198): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '
> [Line 205:80](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/nest2pos.py#L205): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (89 > 79 characters)

* In the file [`cpnest/sampler.py`](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py):

> [Line 69:22](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L69): [E221](https://duckduckgo.com/?q=pep8%20E221) multiple spaces before operator
> [Line 69:22](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L69): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 69:32](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L69): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 70:24](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L70): [E221](https://duckduckgo.com/?q=pep8%20E221) multiple spaces before operator
> [Line 70:24](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L70): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 70:32](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L70): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 71:25](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L71): [E221](https://duckduckgo.com/?q=pep8%20E221) multiple spaces before operator
> [Line 71:25](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L71): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 71:32](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L71): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 72:29](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L72): [E221](https://duckduckgo.com/?q=pep8%20E221) multiple spaces before operator
> [Line 72:29](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L72): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 72:32](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L72): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 73:30](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L73): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 73:32](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L73): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 74:26](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L74): [E221](https://duckduckgo.com/?q=pep8%20E221) multiple spaces before operator
> [Line 74:26](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L74): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 74:32](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L74): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 75:26](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L75): [E221](https://duckduckgo.com/?q=pep8%20E221) multiple spaces before operator
> [Line 75:26](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L75): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 75:32](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L75): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 76:29](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L76): [E221](https://duckduckgo.com/?q=pep8%20E221) multiple spaces before operator
> [Line 76:29](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L76): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 76:32](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L76): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 77:25](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L77): [E221](https://duckduckgo.com/?q=pep8%20E221) multiple spaces before operator
> [Line 77:25](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L77): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 77:32](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L77): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 106:26](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L106): [E221](https://duckduckgo.com/?q=pep8%20E221) multiple spaces before operator
> [Line 107:25](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L107): [E221](https://duckduckgo.com/?q=pep8%20E221) multiple spaces before operator
> [Line 108:21](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L108): [E221](https://duckduckgo.com/?q=pep8%20E221) multiple spaces before operator
> [Line 108:39](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L108): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 108:80](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L108): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 141:53](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L141): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 189:72](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L189): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 189:80](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L189): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (92 > 79 characters)
> [Line 190:80](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L190): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (97 > 79 characters)
> [Line 192:33](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L192): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 194:80](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L194): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (90 > 79 characters)
> [Line 196:80](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L196): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (88 > 79 characters)
> [Line 196:84](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L196): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 197:33](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L197): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 265:27](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L265): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 386:1](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/cpnest/sampler.py#L386): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace

* In the file [`examples/eggbox.py`](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/examples/eggbox.py):

> [Line 9:10](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/examples/eggbox.py#L9): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 9:15](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/examples/eggbox.py#L9): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 10:11](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/examples/eggbox.py#L10): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 10:15](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/examples/eggbox.py#L10): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 10:27](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/examples/eggbox.py#L10): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 10:30](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/examples/eggbox.py#L10): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 10:43](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/examples/eggbox.py#L10): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 10:43](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/examples/eggbox.py#L10): [E262](https://duckduckgo.com/?q=pep8%20E262) inline comment should start with '# '
> [Line 10:80](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/examples/eggbox.py#L10): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (89 > 79 characters)
> [Line 12:1](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/examples/eggbox.py#L12): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 13:28](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/examples/eggbox.py#L13): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 15:1](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/examples/eggbox.py#L15): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 16:23](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/examples/eggbox.py#L16): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 17:33](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/examples/eggbox.py#L17): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 40:9](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/examples/eggbox.py#L40): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 40:25](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/examples/eggbox.py#L40): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 40:35](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/examples/eggbox.py#L40): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 40:46](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/examples/eggbox.py#L40): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 40:57](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/examples/eggbox.py#L40): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 40:70](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/examples/eggbox.py#L40): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 40:80](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/examples/eggbox.py#L40): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (84 > 79 characters)

* In the file [`examples/test_50d.py`](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/examples/test_50d.py):

> [Line 37:18](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/examples/test_50d.py#L37): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 37:80](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/examples/test_50d.py#L37): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (107 > 79 characters)

* In the file [`tests/test_2d.py`](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/tests/test_2d.py):

> [Line 31:18](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/tests/test_2d.py#L31): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 31:43](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/tests/test_2d.py#L31): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 31:53](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/tests/test_2d.py#L31): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 31:64](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/tests/test_2d.py#L31): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 31:75](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/tests/test_2d.py#L31): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 31:80](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/tests/test_2d.py#L31): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (102 > 79 characters)
> [Line 31:88](https://github.com/johnveitch/cpnest/blob/da9974af20063457853475c8c7f7afa4eb9c7e37/tests/test_2d.py#L31): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','



##### Comment last updated at 2020-05-29 09:30:36 UTC

wdpozzo commented:
Hi  John, this is long overdue. Note that, the proper calculation always  finds extremely low ACLs, order 20.

johnveitch commented on cpnest/sampler.py:
```diff
@@ -122,15 +127,20 @@ def reset(self):
             
             prior_samples = []
-            for k in tqdm(range(self.maxmcmc), desc='SMPLR {} generating prior samples'.format(self.thread_id),
+            for k in tqdm(range(5000), desc='SMPLR {} generating prior samples'.format(self.thread_id),
```
Why 5000? Probably better change this back as it is unrelated to the ACL?

johnveitch commented on cpnest/nest2pos.py:
```diff
@@ -171,14 +171,17 @@ def autocorrelation(x):
     for i in range(len(acf)):
-        if acf[i] < i/5.0:
-            return acf[i]+1
+        if np.abs(acf[i]) < i/safety:
```
The [autocorrelation length used in lalinference](https://git.ligo.org/lscsoft/lalsuite/-/blob/master/lalinference/lib/LALInferenceNestedSampler.c#L1261) is defined differently.
Let 
L = \sum_i^j 2*Cxx[i]
Where Cxx[i] is the autocorrelation at lag i for parameter x.
Then the autocorrelation length is the value of L when Cxx[j]<TOLERANCE, where TOLERANCE is 0.01 in LALInferenceNest.

wdpozzo commented:
Ok, I implemented the changes you suggested. I did not use pearsonr because is extremely slow and gives identical results to just computing the whole acf with an FFT. Also, i implemented the AR method, that is commented out. it is unfeasibly slow, so it can be removed if  you want to.

johnveitch commented:
Thanks Walter. I made a few minor tweaks. The HMC 2d test is failing with
```
john@athena:~/src/cpnest$ python3 tests/test_2d_hmc.py
Checkpoint exception raise
test_run (__main__.GaussianTestCase) ... 2020-05-28, 10:06:34 - CPNest  : Setting verbosity to 1
2020-05-28, 10:06:34 - CPNest  : Running with 1 parallel threads
Process Process-2:
Traceback (most recent call last):                                                                           | 0/1000 [00:00<?, ?it/s]
  File "/usr/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/home/john/src/cpnest/cpnest/sampler.py", line 202, in produce_sample
    self._produce_sample()
  File "/home/john/src/cpnest/cpnest/sampler.py", line 214, in _produce_sample
    self.reset()
  File "/home/john/src/cpnest/cpnest/sampler.py", line 138, in reset
    _, p = next(self.yield_sample(-np.inf))
  File "/home/john/src/cpnest/cpnest/sampler.py", line 393, in yield_sample
    p.update_trajectory_length(self.estimate_nmcmc())
  File "/home/john/src/cpnest/cpnest/sampler.py", line 190, in estimate_nmcmc
    ACL = [acl(samples[:,i]) for i in range(samples.shape[1])]
IndexError: tuple index out of range
```

johnveitch commented on cpnest/sampler.py:
```diff
@@ -167,7 +177,27 @@ def estimate_nmcmc(self, safety=5, tau=None):
+        else:
+            self.ACL = [acl(samples[-self.maxmcmc:,i]) for i in range(samples.shape[1])]
+        self.Nmcmc = int(np.max(self.ACL))
+        if self.store_chain is False and len(self.samples) > self.maxmcmc:
```
I don't understand the logic behind this. Can you add a comment?

wdpozzo replied:
done.  the idea is that if you are storing the full mcmc chain, then you should use  only the last maxmcmc samples to compute the ACL and not the full story

johnveitch commented:
Thanks Walter. I am still seeing a problem with the HMC test, it's failing a bit later though. Do you see the same thing?

wdpozzo commented:
The last commit should fix it.

johnveitch commented:
Thanks. I am seeing an error when running with verbose=3 (in the examples/eggbox.py program). It looks like it's unable to stack prior samples when preparing to plot:
```
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
~/src/cpnest/examples/eggbox.py in <module>
     39     M = EggboxModel()
     40     work=cpnest.CPNest(M,verbose=3,nthreads=4,nlive=1000,maxmcmc=1000,poolsize=1000)
---> 41     work.run()
     42

~/src/cpnest/cpnest/cpnest.py in run(self)
    224             sys.exit(130)
    225
--> 226         self.posterior_samples = self.get_posterior_samples(filename=None)
    227         if self.verbose>1: self.plot(corner = False)
    228

~/src/cpnest/cpnest/cpnest.py in get_posterior_samples(self, filename)
    292
    293             # first deal with the prior samples
--> 294             self.prior_samples = stack_arrays([p for p in prior_samples])
    295             if filename:
    296                 np.savetxt(os.path.join(

<__array_function__ internals> in stack_arrays(*args, **kwargs)

/usr/lib/python3/dist-packages/numpy/lib/recfunctions.py in stack_arrays(arrays, defaults, usemask, asrecarray, autoconvert)
   1300     fldnames = [d.names for d in ndtype]
   1301     #
-> 1302     dtype_l = ndtype[0]
   1303     newdescr = _get_fieldspec(dtype_l)
   1304     names = [n for n, d in newdescr]

IndexError: list index out of range
```

I know it's not caused by these changes but is there any chance you could take a look?

wdpozzo commented:
wow that is a big job! the code is full  of inconsistencies  for prior_sampling and verbose. will try and fix it in the next few hours
